crypto/azure/keyvault/component.go

```diff
@@ -0,0 +1,418 @@
+	}
+
+	if res.Result == nil {
+		return nil, nil, errors.New("response from Key Vault does not contain a valid wrapped key")
+	}
+
+	return res.Result, res.AuthenticationTag, nil
+}
+
+// UnwrapKey unwraps a key.
+// The key argument can be in the format "name" or "name/version".
+func (k *keyvaultCrypto) UnwrapKey(parentCtx context.Context, wrappedKey []byte, algorithmStr string, key string, nonce []byte, tag []byte, associatedData []byte) (plaintextKey jwk.Key, err error) {
+	kid := newKeyID(key)
+
+	algorithm := GetJWKEncryptionAlgorithm(algorithmStr)
+	if algorithm == nil {
+		return nil, fmt.Errorf("invalid algorithm: %s", algorithmStr)
+	}
+
+	ctx, cancel := context.WithTimeout(parentCtx, requestTimeout)
+	res, err := k.vaultClient.UnwrapKey(ctx, kid.Name, kid.Version, azkeys.KeyOperationsParameters{
+		Algorithm: algorithm,
+		Value:     wrappedKey,
+		IV:        nonce,
+		Tag:       tag,
+		AAD:       associatedData,
+	}, nil)
+	cancel()
+	if err != nil {
+		return nil, fmt.Errorf("error from Key Vault: %w", err)
+	}
+
+	if res.Result == nil {
+		return nil, errors.New("response from Key Vault does not contain a valid unwrapped key")
+	}
+
+	// Key Vault allows wrapping/unwrapping only symmetric keys, so no need to try and decode an ASN.1 DER-encoded sequence
+	plaintextKey, err = jwk.FromRaw(res.Result)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create JWK from raw key: %w", err)
+	}
+
+	return plaintextKey, nil
+}
+
+// Sign a digest.
+// The key argument can be in the format "name" or "name/version".
+func (k *keyvaultCrypto) Sign(parentCtx context.Context, digest []byte, algorithmStr string, key string) (signature []byte, err error) {
+	kid := newKeyID(key)
+
+	algorithm := GetJWKSignatureAlgorithm(algorithmStr)
+	if algorithm == nil {
+		return nil, fmt.Errorf("invalid algorithm: %s", algorithmStr)
+	}
+
+	ctx, cancel := context.WithTimeout(parentCtx, requestTimeout)
+	res, err := k.vaultClient.Sign(ctx, kid.Name, kid.Version, azkeys.SignParameters{
+		Algorithm: algorithm,
+		Value:     digest,
+	}, nil)
+	cancel()
+	if err != nil {
+		return nil, fmt.Errorf("error from Key Vault: %w", err)
+	}
+
+	if res.Result == nil {
+		return nil, errors.New("response from Key Vault does not contain a valid signature")
+	}
+
+	return res.Result, nil
+}
+
+// Verify a signature.
+// The key argument can be in the format "name" or "name/version".
+func (k *keyvaultCrypto) Verify(parentCtx context.Context, digest []byte, signature []byte, algorithmStr string, key string) (valid bool, err error) {
+	kid := newKeyID(key)
+
+	algorithm := GetJWKSignatureAlgorithm(algorithmStr)
+	if algorithm == nil {
+		return false, fmt.Errorf("invalid algorithm: %s", algorithmStr)
+	}
+
+	// Verifying with non-cacheable keys must happen in the vault
+	if !kid.Cacheable() {
+		return k.verifyInVault(parentCtx, digest, signature, algorithm, kid)
+	}
+
+	// Using a cacheable, asymmetric key, we can verify the data directly here
```

Comment:
Azure Key Vault only supports asymmetric algorithms for digital signatures (and `GetJWKSignatureAlgorithm` enforces that). A mismatch between key type and algorithm would cause an error in the vault.